### ISSUE TITLE
Error on Mathenpoche (french school mathematics)

### ISSUE THREAD

**Issue by Tagadacouincoin** [ruffle-rs/ruffle#2693]
Hello, 

In french school, we used many times : 
http://archives.mathenpoche.net/6eme/pages/menu.html

I try to install it on my website with ruffle and i find many errors. 

First one : 

https://lesmathematiquescpasautomatique.fr/wp-content/wp-mep/6eme/pages/numerique/chap1/serie1/menuexo.swf

When i try on my website or on demomode : 
whats appear : 

![image](https://user-images.githubusercontent.com/77745264/105216506-f1605400-5b52-11eb-871c-5faf03b63baf.png)

But without ruffle this is that : 

![image](https://user-images.githubusercontent.com/77745264/105216847-63d13400-5b53-11eb-8f0a-390647ca2758.png)

Many thanks for this projekt ! 

(Sorry for my language... not really good in english ^^")

**Comment by Toad06** [comment]
Thanks for the report.

The issue is that `loadVariables` calls don't seem to be synchronous in Ruffle, so code execution continues even if variables are not loaded yet. In this sample, [titres.txt](http://archives.mathenpoche.net/6eme/pages/numerique/chap1/serie1/titres.txt) appears to be loaded too late.

In `\frame 1\PlaceObject2 (79) Depth: 10 (clipo)\onClipEvent(load)`:
```
loadVariables("titres.txt","_root.clipo","GET");
i = 1;

trace(_root.clipo.nombre_exos); // undefined in Ruffle
setTimeout(function() {
   trace(_root.clipo.nombre_exos); // 7 in Ruffle
}, 1000);

while(i <= _root.clipo.nombre_exos)
{
   _root.clipo.tritre[i] = unescape(_root.clipo.titre[i]);
   i++;
}
```

**Comment by Herschel** [comment]
`loadVariables` is asynchronous (in your example `trace(_root.clipo.nombre_exos);` will trace undefined initially in Flash Player as well). The issues is we don't yet implement `onClipEvent(data)`/`onData`, which should be fired when variables are loaded into a clip.

**Comment by Toad06** [comment]
The main issue is now fixed (`onClipEvent(data)`/`onData` is fired, so the buttons are visible - cc. #7311).

However hovering a button won't display the expected value (always "Exercice 1" - this might be the same issue as #2181).

**Comment by n0samu** [comment]
The hover text issue has been fixed by #12541.
